Achievement recording fixes

NHTangles opened the pull request:
This properly fixes the issues with mines and soko achievement recording, as follows:
1. record_achieve_special is mapped to corpsenm, which has a default value of -1.  Setting to 1 on the prize item and then checking for nonzero will always mark the achievement (current workaround is to also check the level at pickup time, but this has other issues outlined below).
2. only checking level and otyp when setting record_achieve_special opens the possibility that a random luckstone may be generated at mine-end, and this may be interpreted as the prize (this has possibly happened though it's hard to be conclusive as monsters pick things up and move them)
3. checking the level at pickup time is problematic if a monster picks up the prize and takes it to another level before the hero gets his hands on it (cases of this have been recorded).

NHTangles committed:
Achievement recording fixes (soko/mines).

1. Set record_achieve_special flag only if item is not ramdom
 - This prevents achievement on picking up random luckstone at mineend
2. Explicitly check record_achieve_special == 1 on pickup.
 - Default value is -1 so checking for nonzero doesn't cut it.
3. Do not check level at pickup time. Correctly checking item and flag
is sufficient, and prevents issues when a monster picks up the prize
and takes it off the level.

FredrIQ commented:
What if the player picks up the luckstone, walks somewhere else, and is killed, and someone else finds the bones? Wouldn't your fixes make the achievement count for them? Your changes doesn't seem to address this (bones.c isn't changed)

NHTangles commented:
Thanks for looking, FIQ.
This case is taken care of in invent.c - The player picks up the object, the achievement is recorded, and then the record_achieve_special flag is zeroed, so the object no longer has any special significance to the player, or any subsequent players that find it in bones.

NHTangles committed:
Remove achivement flags in bones

Just in case the mines/soko prizes get to a bones-eligible level
without the hero picking them up, and the hero dies on that level
and leaves bones, we clear the flag to avoid false achievements
being recorded for future players who find the bones.

NHTangles commented:
After further discussion with FIQ, it would indeed be possible for a prize item to get to another level without the player picking it up (eg by a pet or another monster) and potentially be left in bones.  Added another commit just now to address this.

Xaleth reviewed (changes requested):
Remove unnecessary space in line 185 in bones.c

NHTangles commented:
Closing this because the issues have been otherwise addressed in the following commits:
58477b33f4e797fab8c8e5202e146a678de448d5
5a50bb64451196433380cd70a884f7791ce2a288
